docs/macios/interpreter.md

```diff
@@ -0,0 +1,82 @@
+
+However, there is a security restriction on iOS, set by Apple, which disallows the execution of dynamically generated code on a device. Similarly, the execution of dynamically generated code is disallowed on Mac Catalyst apps running on the ARM64 architecture. To meet this restriction, iOS and Mac Catalyst apps use an Ahead of Time (AOT) compiler to compile the managed code. This produces a native iOS binary that can be deployed to Apple devices, or a native Mac Catalyst binary.
+
+AOT provides benefits through a reduction in startup time, and various other performance optimizations. However, it also restricts certain features from being used in your app:
+
+- There's limited generics support. Not every possible generic instantiation can be determined at compile time. Many of the iOS-specific issues encountered in .NET MAUI release builds are due to this limitation.
+- Dynamic code generation isn't allowed. This means that `System.Relection.Emit` is unavailable, there's no support for `System.Runtime.Remoting`, and some uses of the C# [dynamic](/dotnet/csharp/advanced-topics/interop/using-type-dynamic) type aren't permitted.
+
+The Mono interpreter overcomes these restrictions while abiding by platform restrictions. It enables you to interpret some parts of your app at runtime, while AOT compiling the rest. However, there are some drawbacks to using the interpreter in a production app:
+
+- While the app size usually shrinks significantly when the interpreter is enabled, in certain cases the app size can increase.
+- App execution speed will be slower because interpreted code runs more slowly than AOT compiled code. This execution speed reduction can range from unmeasurable to unacceptable, so performance testing should be performed.
+- Native stack traces in crash reports become less useful, because they'll contain generic frames from the interpreter that don't mention the code that's executing. However, managed stack traces won't change.
+
+The interpreter is enabled by default for .NET MAUI debug builds, and can be enabled for release builds.
+
+> [!TIP]
+> If your .NET MAUI iOS app or ARM64-based Mac Catalyst app works correctly as a debug build but then crashes as a release build, try enabling the interpreter for your app's release build. It may be that your app, or one of its libraries, uses a feature that requires the interpreter.
+
+## Enable the interpreter
+
+The Mono interpreter can be enabled in release builds by setting the `UseInterpreter` MSBuild property to `true` in your .NET MAUI app's project file:
+
+```xml
+<PropertyGroup Condition="$(TargetFramework.Contains('-ios')) and '$(Configuration)' == 'Release'">
+    <UseInterpreter>true</UseInterpreter>
+</PropertyGroup>
+```
+
+The interpreter can be enabled for Mac Catalyst release builds on ARM64 by changing `-ios` in the above example to `-maccatalyst-arm64`.
```

Comment:
```suggestion
The interpreter can be enabled for Mac Catalyst release builds on ARM64 by changing `-ios` in the above example to `-maccatalyst`.
```

If you want to condition on arm64, you need to do something like this:

```xml
<PropertyGroup Condition="$(RuntimeIdentifier)' == 'maccatalyst-arm64' and '$(Configuration)' == 'Release'">
    <UseInterpreter>true</UseInterpreter>
</PropertyGroup>
```

That said, setting `UseInterpreter` on x64 is harmless, so it doesn't really matter.